inspire_disambiguation/api.py

```diff
@@ -221,30 +210,137 @@ def cluster(
             ),
             signature_block=phonetic_block,
             B3_f_score=cluster.supervised_scoring(clusterer.y, cluster.labels_)
-            if hasattr(cluster, "supervised_scoring") and not test_labels
+            if hasattr(cluster, "supervised_scoring")
             else None,
         )
-    if test_labels:
-        statistics_names = ("precision", "recall", "f1")
-        training_statistics = clusterer.score(test_signatures_uuids, test_labels)
-        (
-            (B3_statistics_all, wrongly_classified_pairs),
-            B3_statistics_training,
-            B3_statistics_test,
-        ) = training_statistics
+    return process_clustering_output(clusterer)
+
+
+def cluster_with_evaluation(
+    ethnicity_model_path, distance_model_path, n_jobs, test_signatures_uuids=None,
+):
+    """Train the clustering model and process the output.
+
+    Args:
+        ethnicity_model_path (str): Full path where ethnicity model is saved.
+        distance_model_path (str): Full path where distance model is saved.
+        n_jobs (int): Number of processes to use.
+        test_signatures_uuids (set): Signature uuids which will be used
+        for model validation.
+    """
+    start_time = datetime.now()
+    LOGGER.info("Preparing test dataset...")
+    signature_blocks = get_signature_blocks(only_curated=True)
+    labels_train, labels_test, y_train, y_test = (
+        np.array([]),
+        np.array([]),
+        np.array([]),
+        np.array([]),
```

Comment:
Minor comment: do you know in advance how big these arrays will be? if so, it's better to zero them rather than grow the array (as growing an array requires allocating + copying).

Reply:
No, I don't know that. I can do the query ES for all the signatures but I do multiple requests afterwards, so there is the dataset size might change and I'll get the error. I think that this solution is not the most efficient but safer.